libqrexec/qrexec.h

```diff
@@ -23,9 +23,15 @@
 
 #include <stdint.h>
 
-#define QREXEC_PROTOCOL_VERSION 2
+#define QREXEC_PROTOCOL_VERSION 3
 #define MAX_FDS 256
-#define MAX_DATA_CHUNK 4096
+/* protocol version 2 */
+#define MAX_DATA_CHUNK_V2 4096
+/* protocol version 3+ */
+#define MAX_DATA_CHUNK_V3 65536
+
+/* large, but arbitrary */
+#define MAX_SERVICE_NAME_LEN 65536
```

Comment:
Probably doesn't matter but why not a bit smaller such that the message will fit one chunk?

Reply:
Good idea.